### ISSUE TITLE
Fails 2fa by retrying with your last code

### ISSUE THREAD

**Issue by Alexgopen** [SteamRE/DepotDownloader#309]
I've written an install script that requires downloading from two depots.  The first time depot downloader is called, it allows me to sign in and enter my 2fa code.  After it completes all of its downloads, the second call to depot downloader is made, but rather than allow me to sign in and enter 2fa again, it signs in automatically, but then attempts to use the same old 2fa code from before the start of the first set of downloads, which is now long since expired.  It then fails to connect to steam because of 2fa failure.

tl;dr it retries authentication with an expired 2fa code and fails, rather than allowing the user to enter the current 2fa code.

**Comment by Alexgopen** [comment]
![image](https://user-images.githubusercontent.com/16151797/156116761-1119f553-4c8f-49d9-afec-3c1e71cff11f.png)

**Comment by Alexgopen** [comment]
Discovered the cause, it actually wasn't remembering the old 2fa code, rather Console.ReadLine() here was somehow getting data from stdin and returning immediately.  Flushing stdin before this call should fix it
https://github.com/SteamRE/DepotDownloader/blob/e45c8c3d21a43f7b5cae30fc4472aa8eedea93b2/DepotDownloader/Steam3Session.cs#L602

**Comment by azuisleet** [comment]
The most obvious case should be resolved with 73d46d7774a4b9ce0d9262e5e6d8c6a2582c934a. I'm not sure any other data should be discarded.
